llvm/lib/Target/NVPTX/NVPTXCtorDtorLowering.cpp

```diff
@@ -32,6 +33,11 @@ static cl::opt<std::string>
               cl::desc("Override unique ID of ctor/dtor globals."),
               cl::init(""), cl::Hidden);
 
+static cl::opt<bool>
+    CreateKernels("nvptx-lower-global-kernel",
```

Comment:
Naming is hard...
"nvptx-lower-global-kernel" is shorter, but still sounds odd to me. What *is* a global kernel? And what does it mean to "lower" it. While you and I have enough context to make sense of it all, it will not be the case for the end user who would have no clue what the heck is lowering and what does it have to do with a global kernel, whatever that is.

The key here, IMO, is that it's a knob to "make global initializers work", and that's what the name should reflect. It does not need to describe *how* we're going to do it.

Reply:
Honestly the only reason I have this here is because in `libc` I do this manually and figured it would help to turn off some unnecessary work in those cases. However, I can remove the option since it's not *really* required for anything.

Maybe `nvptx-emit-init-fini-kernel`?

Reply:
`nvptx-emit-init-fini-kernel` works for me.